ironfish/src/rpc/adapters/tests/adapterTest.ts

```diff
@@ -2,77 +2,84 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import * as yup from 'yup'
-import { Assert } from '../../assert'
-import { IronfishSdk } from '../../sdk'
-import { getUniqueTestDataDir } from '../../testUtilities'
-import { PromiseUtils } from '../../utils/promise'
-import { RpcRequestError } from '../clients'
-import { RpcIpcClient } from '../clients/ipcClient'
-import { ALL_API_NAMESPACES } from '../routes/router'
-import { RPC_ERROR_CODES, RpcValidationError } from './errors'
-import { RpcIpcAdapter } from './ipcAdapter'
-
-describe('IpcAdapter', () => {
-  let ipc: RpcIpcAdapter
-  let sdk: IronfishSdk
-  let client: RpcIpcClient
+import { FullNode } from '../../../node'
+import {
+  IRpcAdapter,
+  Router,
+  RPC_ERROR_CODES,
+  RpcClient,
+  RpcRequestError,
+  RpcValidationError,
+} from '../../../rpc'
+import { IronfishSdk } from '../../../sdk'
+import { createNodeTest } from '../../../testUtilities'
+import { PromiseUtils } from '../../../utils/promise'
+
+export function createAdapterTest(
+  onSetup: (
+    sdk: IronfishSdk,
+    node: FullNode,
+  ) => Promise<{
+    client: RpcClient
+    router: Router
+    adapter: IRpcAdapter
+  }>,
+  onTeardown: () => void | Promise<void>,
+  onConnect: () => void | Promise<void>,
+  onDisconnect: () => void,
+): void {
+  let client: RpcClient
+  let router: Router
+  let adapter: IRpcAdapter
+
+  const nodeTest = createNodeTest(false, {
+    config: {
+      enableRpc: false,
+      enableRpcIpc: false,
+      enableRpcTcp: false,
+      enableRpcTls: false,
+      enableRpcHttp: false,
+    },
+  })
 
   beforeEach(async () => {
-    sdk = await IronfishSdk.init({
-      dataDir: getUniqueTestDataDir(),
-      configOverrides: {
-        enableRpc: false,
-        enableRpcIpc: false,
-      },
-    })
-
-    ipc = new RpcIpcAdapter(sdk.config.get('ipcPath'), undefined, ALL_API_NAMESPACES)
-
-    const node = await sdk.node()
-    await node.rpc.mount(ipc)
-
-    Assert.isInstanceOf(sdk.client, RpcIpcClient)
-    client = sdk.client
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({ client, router, adapter } = await onSetup(nodeTest.sdk, nodeTest.node))
```

Comment:
Whats the point of the destructuring here?

Why can't this just be:
`await onSetup(nodeTest.sdk, nodeTest.node))`

Reply:
These are destructuring assignments to the state test which are then later used in the tests themselves. These are the variables that are assigned to, https://github.com/iron-fish/ironfish/pull/4585/files#diff-2136dc569defbe82be694ae8ec9ca5b0b190111e49ad68a72a7cdda4fe6b2481R31-R33